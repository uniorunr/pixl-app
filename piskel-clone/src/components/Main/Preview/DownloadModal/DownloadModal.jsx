import React, { Component } from 'react';
import './DownloadModal.scss';
import PropTypes from 'prop-types';

class DownloadModal extends Component {
  state = {
    name: null,
    size: null,
  };

  handleNameInput = ({ target: { value } }) => {
    this.setState({
      name: value,
    });
  };

  handleSizeInput = ({ target: { value } }) => {
    this.setState({
      size: value,
    });
  };

  handleDownload = () => {
    const { name, size } = this.state;
    const { download } = this.props;
    download(name, size);
  };

  render() {
    const { closeDownloadModal } = this.props;

    return (
      <section className="download-modal">
        <div className="download-modal__content">
          <span>File Name (optional):</span>
          <input
            className="download-modal__file-name"
            type="text"
            onInput={this.handleNameInput}
          />
          <span>Size (optional):</span>
          <input
            className="download-modal__size"
            type="text"
            onInput={this.handleSizeInput}
          />
          <button
            type="button"
            className="download-modal__submit"
            onClick={this.handleDownload}
          >
            Download
          </button>
        </div>
        <button
          className="download-modal__close"
          type="button"
          onClick={closeDownloadModal}
        >
          <span>x</span>
        </button>
      </section>
    );
  }
}

DownloadModal.propTypes = {
  closeDownloadModal: PropTypes.func.isRequired,
  download: PropTypes.func.isRequired,
};

export default DownloadModal;
