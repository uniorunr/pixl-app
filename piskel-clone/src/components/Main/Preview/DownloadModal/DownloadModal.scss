.download-modal {
  border: 2px solid black;
  margin: auto;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  width: 97%;
  height: 100%;
  position: absolute;
  background: white;
  z-index: 1;
  display: flex;
  justify-content: center;
  align-items: center;

  .download-modal__content {
    display: flex;
    flex-direction: column;

    input {
      max-width: 100px;
      margin: 10px 0;
      padding: 2px 5px;
      border: 2px solid black;
      outline: none;
    }

    input:focus {
      border: 2px dashed black;
    }

    button {
      position: relative;
      top: 0;
      padding: 5px 10px;
      background-color: white;
      border: 2px solid black;
      box-shadow: 2px 2px 0 black;
      cursor: pointer;
      outline: none;
      font-weight: 700;
      font-family: "Roboto", "Arial", sans-serif;
      font-size: 1.4rem;
    }

    button:active {
      top: 2px;
      box-shadow: 0 0 0 black;
    }
  }

  .download-modal__close {
    position: absolute;
    top: -2px;
    right: -2px;
    border: 2px solid black;
    outline: none;
    background: white;
    cursor: pointer;

    span {
      font-weight: 700;
    }
  }

  .download-modal__close:hover {
    border: 2px dashed black;

    span {
      font-weight: 700;
    }
  }
}
