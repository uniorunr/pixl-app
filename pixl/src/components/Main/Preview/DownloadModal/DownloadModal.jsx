import React, { Component } from 'react';
import './DownloadModal.scss';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import GIF from 'gif.js.optimized';
import download from 'downloadjs';
import * as actions from '../../../../actions/actions';

class DownloadModal extends Component {
  handleNameInput = ({ target: { value } }) => {
    const { setDownloadName } = this.props;
    setDownloadName(value);
  };

  handleSizeInput = ({ target: { value } }) => {
    const { setDownloadSize } = this.props;
    setDownloadSize(+value);
  };

  handleDownload = () => {
    const { name, size } = this.props;
    this.download(name, size);
  };

  download = (returnedName, returnedSize) => {
    const {
      name,
      size,
      updateCurrPreviewFrame,
      updateRecordingState,
      setGifInstance,
      setDownloadModalState,
    } = this.props;
    const finalName = returnedName || name;
    const finalDimensions = returnedSize || size;
    const gif = new GIF({
      workers: 2,
      quality: 10,
      width: finalDimensions,
      height: finalDimensions,
    });

    gif.on('finished', (blob) => {
      download(blob, `${finalName}.gif`, 'image/gif');
    });
    updateCurrPreviewFrame(0);
    updateRecordingState(true);
    setGifInstance(gif);
    setDownloadModalState(false);
  };

  render() {
    const { closeDownloadModal } = this.props;

    return (
      <section className="download-modal">
        <div className="download-modal__content">
          <span>File Name (optional):</span>
          <input
            className="download-modal__file-name"
            type="text"
            onInput={this.handleNameInput}
          />
          <span>Size (optional):</span>
          <input
            className="download-modal__size"
            type="text"
            onInput={this.handleSizeInput}
          />
          <button
            type="button"
            className="download-modal__submit"
            onClick={this.handleDownload}
          >
            Download
          </button>
        </div>
        <button
          className="download-modal__close"
          type="button"
          onClick={closeDownloadModal}
        >
          <span>x</span>
        </button>
      </section>
    );
  }
}

DownloadModal.propTypes = {
  closeDownloadModal: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  size: PropTypes.number.isRequired,
  setGifInstance: PropTypes.func.isRequired,
  updateCurrPreviewFrame: PropTypes.func.isRequired,
  updateRecordingState: PropTypes.func.isRequired,
  setDownloadModalState: PropTypes.func.isRequired,
  setDownloadName: PropTypes.func.isRequired,
  setDownloadSize: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  name: state.preview.name,
  size: state.preview.size,
});

const mapDispatchToProps = (dispatch) => {
  const {
    updatePixelsPerCanvas,
    updateCurrPreviewFrame,
    setGifInstance,
    setDownloadName,
    setDownloadSize,
    updateRecordingState,
    setDownloadModalState,
  } = bindActionCreators(actions, dispatch);
  return {
    updatePixelsPerCanvas,
    updateRecordingState: (state) => {
      updateRecordingState(state);
    },
    setGifInstance: (gif) => {
      setGifInstance(gif);
    },
    updateCurrPreviewFrame: (index) => {
      updateCurrPreviewFrame(index);
    },
    setDownloadName: (name) => {
      setDownloadName(name);
    },
    setDownloadSize: (size) => {
      setDownloadSize(size);
    },
    setDownloadModalState: (state) => {
      setDownloadModalState(state);
    },
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(DownloadModal);
