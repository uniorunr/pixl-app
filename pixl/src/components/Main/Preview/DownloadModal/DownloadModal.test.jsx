import React from 'react';
import renderer from 'react-test-renderer';
import DownloadModal from './DownloadModal';

it('renders correctly', () => {
  const tree = renderer
    .create(<DownloadModal closeDownloadModal={() => {}} download={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
