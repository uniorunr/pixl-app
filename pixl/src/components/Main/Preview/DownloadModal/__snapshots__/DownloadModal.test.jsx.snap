// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="download-modal"
>
  <div
    className="download-modal__content"
  >
    <span>
      File Name (optional):
    </span>
    <input
      className="download-modal__file-name"
      onInput={[Function]}
      type="text"
    />
    <span>
      Size (optional):
    </span>
    <input
      className="download-modal__size"
      onInput={[Function]}
      type="text"
    />
    <button
      className="download-modal__submit"
      onClick={[Function]}
      type="button"
    >
      Download
    </button>
  </div>
  <button
    className="download-modal__close"
    onClick={[Function]}
    type="button"
  >
    <span>
      x
    </span>
  </button>
</section>
`;
